chain/actors/actor_storagemarket.go

```diff
@@ -342,7 +328,21 @@ func (sma StorageMarketActor) PublishStorageDeals(act *types.Actor, vmctx types.
 			return nil, err
 		}
 
-		err := deals.Set(self.NextDealID, &OnChainDeal{Deal: deal})
+		err := deals.Set(self.NextDealID, &OnChainDeal{
+			PieceRef:           deal.PieceRef,
+			PieceSize:          deal.PieceSize,
+			PieceSerialization: deal.PieceSerialization, // TODO: this isnt needed anymore, right?
```

Comment:
We only have unixfs now, and are about to switch to go-car, so might as well drop it now